File: backend/src/main/kotlin/com/finch/domain/deposit/entity/PaymentMethod.kt
package com.finch.domain.deposit.entity

/**
 * 모의 결제 수단 (featureSpec 3.2). `ck_deposit_method` 와 같은 집합이다.
 *
 * **실제 금전 이동은 없다.** 실제 결제와 유사한 절차를 밟게 하려고 두는 시뮬레이션용 선택지이고,
 * 어느 값을 골라도 처리 경로가 같다.
 */
enum class PaymentMethod {
	VIRTUAL_CARD,
	VIRTUAL_TRANSFER,
}

File: frontend/src/shared/lib/paymentMethod.ts
import type { PaymentMethod } from '@/shared/types/deposit';

/**
 * 결제 수단 표시 이름 (featureSpec §3.2).
 *
 * `shared` 에 두는 이유 — 충전 화면과 매매 내역이 둘 다 쓰는데 feature 끼리는 import 하지
 * 않는다 (컨벤션: 의존 방향 `app → pages → features → shared` 단방향).
 *
 * 서버가 문구를 완성해 주는 에러 `message` 와 달리 이것은 **enum 값의 표시 이름**이라
 * 화면 쪽에 있다. 서버는 `VIRTUAL_CARD` 를 주지 "가상 카드" 를 주지 않는다.
 */
export const PAYMENT_METHOD_LABEL: Record<PaymentMethod, string> = {
  VIRTUAL_CARD: '가상 카드',
  VIRTUAL_TRANSFER: '가상 계좌이체',
};
